core/src/main/kotlin/no/ntnu/beardblaster/spell/SpellExecutor.kt

```diff
@@ -15,35 +15,41 @@ class SpellExecutor {
     private val mitigationSpells: MutableList<MitigationSpell> = ArrayList()
 
     fun addSpell(spell: SpellAction, turn: Int) {
-        LOG.info { "Adding spell ${spell.spell.spellName} cast by ${spell.caster} - turn $turn" }
-        if(spellHistory.containsKey(turn)) {
-            spellHistory[turn]?.add(spell)
-        }
-        spellHistory.putIfAbsent(turn, mutableListOf(spell))
-
-        if (spell.spell.spellHealing > 0) {
-            mitigationSpells.add(HealingOverTime(spell.caster, turn, spell.spell.duration, spell.spell.spellHealing))
-        }
-
-        if (spell.spell.spellMitigation > 0) {
-            mitigationSpells.add(
-                DamageReduction(
-                    spell.caster,
-                    turn + 1, // Damage reduction spells should be activated 1 turn after current
-                    spell.spell.duration,
-                    spell.spell.spellMitigation
+        if(isSpellValidForTurn(spell, turn)) {
```

Comment:
1: This is the only real change,  implemented a function to check if the spell was cast by a wizard already.